core/src/main/java/software/amazon/awssdk/core/async/ByteArrayAsyncRequestBody.java

```diff
@@ -43,20 +45,42 @@ public long contentLength() {
     }
 
     @Override
-    public void subscribe(Subscriber<? super ByteBuffer> subscriber) {
-        subscriber.onSubscribe(
-                new Subscription() {
-                    @Override
-                    public void request(long n) {
-                        if (n > 0) {
-                            subscriber.onNext(ByteBuffer.wrap(bytes));
-                            subscriber.onComplete();
+    public void subscribe(Subscriber<? super ByteBuffer> s) {
+        // As per rule 2.13, we need to throw a `java.lang.NullPointerException` if the `Subscription` is `null`
+        if (s == null) {
```

Comment:
Here 's' is Subscriber?

Reply:
I think the comment is wrong, this is the wording for 2.13:

> Calling onSubscribe, onNext, onError or onComplete MUST return normally except when any provided parameter is null in which case it MUST throw a java.lang.NullPointerException to the caller, for all other situations the only legal way for a Subscriber to signal failure is by cancelling its Subscription. In the case that this rule is violated, any associated Subscription to the Subscriber MUST be considered as cancelled, and the caller MUST raise this error condition in a fashion that is adequate for the runtime environment.

I'll update it

Reply:
And 1.9:

> Publisher.subscribe MUST call onSubscribe on the provided Subscriber prior to any other signals to that Subscriber and MUST return normally, except when the provided Subscriber is null in which case it MUST throw a java.lang.NullPointerException to the caller, for all other situations the only legal way to signal failure (or reject the Subscriber) is by calling onError (after calling onSubscribe).